src/canvas/Canvas.ts

```diff
@@ -786,23 +780,25 @@ export class Canvas extends SelectableCanvas implements CanvasOptions {
    * @param {Event} e Event object fired on mouseup
    */
   __onMouseUp(e: TPointerEvent) {
-    const transform = this._currentTransform;
     this._cacheTransformEventData(e);
-    const isClick = this._isClick;
     this._handleEvent(e, 'up:before');
+
+    const transform = this._currentTransform;
+    const isClick = this._isClick;
     const target = this._target;
+
     // if right/middle click just fire events and return
     // target undefined will make the _handleEvent search the target
     if (checkClick(e, RIGHT_CLICK)) {
       if (this.fireRightClick) {
-        this._handleEvent(e, 'up', RIGHT_CLICK, isClick);
+        this._handleEvent(e, 'up', RIGHT_CLICK);
```

Comment:
While the doc stated that isClick was good only for left button, i do think it works also for right and middle.
Let's check if it works and let's keep it in case